rllib/algorithms/bc/tf/bc_tf_learner.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import Mapping
+
+from ray.rllib.core.learner.learner import (
+    POLICY_LOSS_KEY,
+    LearnerHyperparameters,
+)
+from ray.rllib.core.learner.tf.tf_learner import TfLearner
+from ray.rllib.core.rl_module.rl_module import ModuleID
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_tf
+from ray.rllib.utils.nested_dict import NestedDict
+from ray.rllib.utils.typing import TensorType
+
+_, tf, _ = try_import_tf()
+logger = logging.getLogger(__file__)
+
+
+class BCTfLearner(TfLearner):
+    """Implements tf-specific BC loss logic.
+
+    This class implements the BC loss under `self.compute_loss_for_module()`.
+    """
+
+    @override(TfLearner)
+    def compute_loss_for_module(
+        self,
+        *,
+        module_id: ModuleID,
+        hps: LearnerHyperparameters,
+        batch: NestedDict,
+        fwd_out: Mapping[str, TensorType]
+    ) -> TensorType:
+        if self.module[module_id].is_stateful():
```

Comment:
Since we are adding code here for the recurrent case, can you also include this in the test?
We do the same thing in PPO so there's a chance you can copy some code!

Reply:
Also, please add the same TODO here, which I suggested in the torch equivalent code below.

Reply:
On Artur's comment:
This might be difficult as we will have to create an offline dataset for the recurrent case first (e.g. record a PPO + LSTM run vs `StatelessCartPole`).
Maybe we can do this in another PR, but yes, this would be super useful to have.

Reply:
Alright. Makes sense to have a dataset that shows recurrence to apply recurrence successully. Let's move it to another issue. probably the one @ArturNiederfahrenhorst  has opened (#33640).